examples/kprobe/main.go

```diff
@@ -50,7 +46,7 @@ func main() {
 	// pre-compiled program. Each time the kernel function enters, the program
 	// will increment the execution counter by 1. The read loop below polls this
 	// map value once per second.
-	kp, err := link.Kprobe(fn, objs.KprobeExecve)
+	kp, err := link.Kprobe(link.SysExecve(), objs.KprobeExecve)
```

Comment:
I wonder whether we should move this fallback behaviour into Kprobe, Kretprobe, etc. proper. Otherwise our interface isn't very portable. That is, the user specifies "sys_execve" here, and we discover
* the correct arch prefix
* with a fallback to just the plain name

@ti-mo thoughts?

Reply:
Agreed, this needs to become a part of the API, not a hack in the test suite. That doesn't add much value to the user, as it makes it harder to write tools portable across architectures.

We also need to hash out what the user should expect, as in: do they specify `__x64_sys_execve` and rely on the lib to implicitly fix it up to make it portable? Or do they specify `sys_execve`, and the lib always tries with and without the arch prefix? There might be code out there already specifying `__x64_sys_execve` that will break if we change the rules.

I'd suggest the following:
- If we detect an arch prefix, fix it up to the prefix that belongs to the host. This should work in most cases.
- Then, first, do exactly what the user asked for (`sys_execve`). If the symbol already contains an arch prefix, we can bail out here.
- If that fails, try again with the arch prefix prepended (`__x64_sys_execve`). Syscall symbols are in the minority, so this should avoid the most back-and-forth.

Note: Using tracepoints should be preferred for probing most popular syscalls (execve was only chosen here because it's easy to trigger), but I'm not sure if tracepoints yield the same flexibility in terms of walking memory.

Reply:
I think the can do something simpler. Assuming we get `__x64_sys_execve` as the argument:
* First try with the arch specific prefix added: `__x64___x64_sys_execve`. This will fail.
* Fall back to trying without the arch prefix (necessary pre 4.17 anyways): `__x64_sys_execve`. This will work on new kernels.

We shouldn't try to translate arch prefixes from one arch to another. This not any more or less broken than at the moment and a lot simpler.

Reply:
You might be right. The user might want to attach only to `__x64_sys_execve` and bail out on other architectures. We also wouldn't have to maintain a list of architecture prefixes to strip, reducing maintenance overhead.

For portable probes, the bare syscall symbol (`sys_execve`) must be specified. This is probably the cleanest API contract.